src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -0,0 +1,221 @@
+                0x16, 0x70, 0x46, 0xC4, 0xBC, 0xA5, 0x5C, 0x30, 0x0A, 0x06, 0x08, 0x2A, 0x86, 0x48, 0xCE, 0x3D, 0x04, 0x03, 0x02,
+                0x03, 0x49, 0x00, 0x30, 0x46, 0x02, 0x21, 0x00, 0x92, 0x51, 0x0D, 0x1D, 0x60, 0x21, 0x5F, 0x01, 0x1B, 0xDA, 0x2E,
+                0xA7, 0x95, 0xFE, 0xCB, 0x24, 0x94, 0xEF, 0x90, 0xA6, 0x97, 0x87, 0xAD, 0x0A, 0xFC, 0x20, 0x6C, 0x61, 0x8C, 0x09,
+                0xCC, 0x52, 0x02, 0x21, 0x00, 0xA0, 0x36, 0xB0, 0xAF, 0x83, 0xE3, 0x45, 0x5C, 0x09, 0x38, 0xB9, 0x4B, 0xFF, 0x47,
+                0x75, 0x19, 0x72, 0xB8, 0xDC, 0xBE, 0x71, 0x9C, 0x9F, 0x61, 0xB5, 0x56, 0xEC, 0xB4, 0xEA, 0x17, 0xCC, 0x57 },
+              { 0xE7, 0xEF, 0xEA, 0xC3, 0x33, 0x5C, 0x5F, 0xD0, 0xC3, 0xE6,
+                0x34, 0x52, 0x9F, 0x16, 0x70, 0x46, 0xC4, 0xBC, 0xA5, 0x5C } },
+        };
+
+    size_t paaLookupTableIdx;
+    for (paaLookupTableIdx = 0; paaLookupTableIdx < ArraySize(sPAALookupTable); ++paaLookupTableIdx)
+    {
+        if (skid.data_equal(ByteSpan(sPAALookupTable[paaLookupTableIdx].mSKID)))
+        {
+            break;
+        }
+    }
+
+    VerifyOrReturnError(paaLookupTableIdx < ArraySize(sPAALookupTable), CHIP_ERROR_INVALID_ARGUMENT);
+
+    return CopySpanToMutableSpan(ByteSpan{ sPAALookupTable[paaLookupTableIdx].mPAACertificate }, out_dac_buffer);
+}
+
+class ExampleDACVerifier : public DeviceAttestationVerifier
+{
+public:
+    AttestationVerificationResult
+    VerifyAttestationInformation(const ByteSpan & attestation_info_buffer, const ByteSpan & attestation_challenge_buffer,
+                                 const ByteSpan & attestation_signature_buffer, const ByteSpan & pai_cert_der_buffer,
+                                 const ByteSpan & dac_cert_der_buffer, const ByteSpan & attestation_nonce) override;
+};
+
+AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(const ByteSpan & attestation_info_buffer,
+                                                                               const ByteSpan & attestation_challenge_buffer,
+                                                                               const ByteSpan & attestation_signature_buffer,
+                                                                               const ByteSpan & pai_cert_der_buffer,
+                                                                               const ByteSpan & dac_cert_der_buffer,
+                                                                               const ByteSpan & attestation_nonce)
+{
+    // match DAC and PAI VIDs
+    if (!pai_cert_der_buffer.empty())
+    {
+        VendorId paiVid;
+        VendorId dacVid;
+
+        CHIP_ERROR error = ExtractVIDFromX509Cert(pai_cert_der_buffer, paiVid);
+        VerifyOrReturnError(error == CHIP_NO_ERROR || error == CHIP_ERROR_KEY_NOT_FOUND,
+                            AttestationVerificationResult::kPaiFormatInvalid);
+
+        if (error == CHIP_NO_ERROR)
```

Comment:
Suggest adding a helper variable:

const bool pai_has_vid = (error != CHIP_ERROR_KEY_NOT_FOUND);

Then the if can be `if (pai_has_vid)` which helps to understand the logic more easily.

Reply:
done